Mage/src/main/java/mage/abilities/keyword/AffinityForLandTypeAbility.java

```diff
@@ -40,6 +40,7 @@ public AffinityForLandTypeAbility(final AffinityForLandTypeAbility ability) {
         super(ability);
         this.text = ability.text;
         this.filter = ability.filter.copy();
+        this.landType = ability.landType;
```

Comment:
I've reworked this class's constructor so it doesn't need redundant fields or to create two identical filters.